src/argparser.py

```diff
@@ -25,6 +25,27 @@
 import argparse
 import shutil
 import os
+import glob
+
+def load_avail_arch(omniperf_home):
+    '''
+    Load list of avail archs by checking against defined implementations
+    '''
+    matching_files=glob.glob(os.path.join(omniperf_home, 'omniperf_soc', 'soc_*.py'))
```

Comment:
I'm not crazy about this approach. Assuming I'm grok'ing correctly, this assumes that if a file exists in the `omniperf_soc` directory that follows a particular naming scheme, then it will automatically be added as a "supported" arch.  That doesn't necessarily match reality if these files are not instantiated as such in omniperf_base.py.  What if the file is a work in progress, or is a deprecated SOC in the future?

Thinking about this more, I guess our current authoritative source for supported targets is defined in omniperf_base.py/detect_soc().

Would you agree?

Reply:
Good point. In theory, there could be incomplete SoC definitions in `omniperf_soc`. One of the biggest considerations for an "optimal" solution is minimizing overhead for adding a new SoC. Ideally, the procedure is two steps:

1. Define an instantiation of SoC in `omniperf_soc`
2. Modify detect_soc() to consider your new SoC instance

I suppose we could add a property (a list called **__supported_archs**) to the Omniperf class such that detect_soc() is responsible for setting the value? At least this way the procedure still only requires steps (1) and (2)....